pkg/trait/knative.go

```diff
@@ -454,11 +468,23 @@ func (t *knativeTrait) configureSinkBinding(e *Environment, env *knativeapi.Came
 	}
 
 	err := t.withServiceDo(false, e, env, services, serviceType, knativeapi.CamelEndpointKindSink, func(ref *corev1.ObjectReference, serviceURI string, _ func() (*url.URL, error)) error {
-		e.ApplicationProperties["camel.k.customizer.sinkbinding.enabled"] = "true"
-		e.ApplicationProperties["camel.k.customizer.sinkbinding.name"] = ref.Name
-		e.ApplicationProperties["camel.k.customizer.sinkbinding.type"] = string(serviceType)
-		e.ApplicationProperties["camel.k.customizer.sinkbinding.kind"] = ref.Kind
-		e.ApplicationProperties["camel.k.customizer.sinkbinding.api-version"] = ref.APIVersion
+		if e.ApplicationProperties == nil {
```

Comment:
@lburgazzoli KnativeResource registry binding

Reply:
Once [CAMEL-20400](https://issues.apache.org/jira/browse/CAMEL-20400) will become available, it should be possible to just use properties

```
came.component.knative.sink-binding.name = …
```

it is also possible to use properties to configure the knative environment (which is also the preferred way), like:

```
camel.component.knative.environment.resources[0].type = ...
camel.component.knative.environment.resources[0].name = ...
```